servicetalk-encoding-api/src/main/java/io/servicetalk/encoding/api/CodecDecodingException.java

```diff
@@ -17,7 +17,9 @@
 
 /**
  * Exception thrown when something goes wrong during decoding.
+ * @deprecated Use {@link BufferEncodingException}.
  */
+@Deprecated
 public final class CodecDecodingException extends RuntimeException {
```

Comment:
Same question here, should we make this and `CodecEncodingException` subtypes of `SerializationException`?